provider/resource.go

```diff
@@ -1018,8 +1020,11 @@ func Provider() tfbridge.ProviderInfo {
 					Source: "express_route_circuit_authorization.html.markdown",
 				},
 			},
-			"azurerm_express_route_gateway":          {Tok: azureResource(azureNetwork, "ExpressRouteGateway")},
-			"azurerm_nat_gateway":                    {Tok: azureResource(azureNetwork, "NatGateway")},
+			"azurerm_express_route_gateway": {Tok: azureResource(azureNetwork, "ExpressRouteGateway")},
+			"azurerm_nat_gateway":           {Tok: azureResource(azureNetwork, "NatGateway")},
+			"azurerm_nat_gateway_public_ip_association": {
+				Tok: azureResource(azureNetwork, "NatGatewayPublicIpAssociation"),
```

Comment:
Also `Ip` should in theory be `IP` but we are already inconsistent with it, so I don't have a good answer what to do here.

Reply:
i would suggest we open another issue to track the inconsistencies and then we can fix those up at a later time